src/sparse/impl/KokkosSparse_rcm_impl.hpp

```diff
@@ -0,0 +1,584 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//               KokkosKernels 0.9: Linear Algebra and Graph Kernels
+//                 Copyright 2017 Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Siva Rajamanickam (srajama@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include "KokkosKernels_Utils.hpp"
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Atomic.hpp>
+#include <impl/Kokkos_Timer.hpp>
+#include <Kokkos_Sort.hpp>
+#include <Kokkos_MemoryTraits.hpp>
+#include <Kokkos_Parallel_Reduce.hpp>
+#include "KokkosBlas1_fill.hpp"
+#include "KokkosGraph_Distance1Color.hpp"
+#include "KokkosKernels_Uniform_Initialized_MemoryPool.hpp"
+
+#ifndef _KOKKOSRCMIMP_HPP
+#define _KOKKOSRCMIMP_HPP
+
+namespace KokkosSparse{
+
+namespace Impl{
+
+template <typename HandleType, typename lno_row_view_t, typename lno_nnz_view_t>
+struct RCM
```

Comment:
It would be useful to expose RCM to the users rather than in Impl. Doesn't have to be part this PR though. We could add an issue.